tests/unit/geom/CoordinateArraySequenceFactoryTest.cpp

```diff
@@ -79,7 +79,7 @@ namespace tut
 			ensure( 0 != col);
 
 			const size_t size0 = 0;
-			CoordinateSequencePtr sequence = factory->create(col);
```

Comment:
Default values: The previous signature was with just one integer, i.e.

    CoordinateSequence *create(std::vector<Coordinate> *coords, std::size_t dims)

with `dims` 1, 2 or 3. The new signature is

    CoordinateSequence *create(std::vector<Coordinate> *coords, int hasZ, int hasM)

which, if it provided default values, would be compatible with the old signature, but the user would end up potentially passing `3` to `hasZ` which is undesirable. Hence breaking the API forces the user to adapt the calls correctly.

Why `int` and not `bool`: the value `-1` is needed if the dimensions are unknown (i.e. what before was `dimension=0`. If `-1`, the dimensions are autodetected as they were before, i.e. in

    CoordinateArraySequence::getHasZ() const
    CoordinateArraySequence::getHasM() const

I suppose a way to retain API compatibility would be something like

    CoordinateSequence *create(std::vector<Coordinate> *coords, std::size_t dims, int dim3isM)

with `dim3isM` accepting -1 (autodetect), 0 (false) or 1 (true). `dims` would then accept 1, 2, 3 or 4 (and `dim3isM` relevant only if `dims=3`). Perhaps this would be a better approach?